code/modules/reagents/reagent_containers/jug.dm

```diff
@@ -0,0 +1,54 @@
+//Not to be confused with /obj/item/reagent_containers/food/drinks/bottle
+
+
+/obj/item/reagent_containers/glass/jug
+	name = "Jug"
+	desc = "A decent sized plastic jug."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "plastic_jug"
+	item_state = "plastic_jug"
+	amount_per_transfer_from_this = 10
+	possible_transfer_amounts = list(1,2,5,10,20,40,80)
+	container_type = OPENCONTAINER
+	volume = 80
+
+/obj/item/reagent_containers/glass/jug/New()
+	..()
+	add_lid()
+
+/obj/item/reagent_containers/glass/jug/proc/add_lid()
```

Comment:
I'd say that it would be best to put this on `/obj/item/reagent_containers` level. Making a `open_lid` proc there will also be nice. You can copy the logic from `/obj/item/reagent_containers/attack_self` then. Just be sure to check if a user does it or not.

Reply:
I wrote that proc exclusively for jugs, since I wanted them to start with their lids on. There is already code in reagent_containers.dm that handles removing/adding lids.

Reply:
Indeed so yeah. It would be best to make the lid adding and removal a proc on it's own on the `reagent_containers` object. Instead of part of the attack_self. This way other containers can also use the code and then you only have one way to open/close lids